### PR TITLE
TST closes #28980 Certain comparison operations misbehaving for period dtype

### LINKED ISSUE #28980: Certain comparison operations misbehaving for period dtype
This is closely related to https://github.com/pandas-dev/pandas/issues/28930.

```python
import pandas as pd

s = pd.Series([pd.Period("2019"), pd.Period("2020")], dtype="period[A-DEC]") 
s == "a"                                                                                                                                                     
```

yields

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-4-3a654234a428> in <module>
----> 1 s == "a"

~/pandas/pandas/core/ops/__init__.py in wrapper(self, other)
    527         rvalues = extract_array(other, extract_numpy=True)
    528 
--> 529         res_values = comparison_op(lvalues, rvalues, op)
    530 
    531         return _construct_result(self, res_values, index=self.index, name=res_name)

~/pandas/pandas/core/ops/array_ops.py in comparison_op(left, right, op)
    253 
    254     if should_extension_dispatch(lvalues, rvalues):
--> 255         res_values = dispatch_to_extension_op(op, lvalues, rvalues)
    256 
    257     elif is_scalar(rvalues) and isna(rvalues):

~/pandas/pandas/core/ops/dispatch.py in dispatch_to_extension_op(op, left, right, keep_null_freq)
    134 
    135     try:
--> 136         res_values = op(left, right)
    137     except NullFrequencyError:
    138         # DatetimeIndex and TimedeltaIndex with freq == None raise ValueError

~/pandas/pandas/core/arrays/period.py in wrapper(self, other)
     98             result.fill(nat_result)
     99         else:
--> 100             other = Period(other, freq=self.freq)
    101             result = ordinal_op(other.ordinal)
    102 

~/pandas/pandas/_libs/tslibs/period.pyx in pandas._libs.tslibs.period.Period.__new__()
   2459                 value = str(value)
   2460             value = value.upper()
-> 2461             dt, _, reso = parse_time_string(value, freq)
   2462             if dt is NaT:
   2463                 ordinal = NPY_NAT

~/pandas/pandas/_libs/tslibs/parsing.pyx in pandas._libs.tslibs.parsing.parse_time_string()
    265             yearfirst = get_option("display.date_yearfirst")
    266 
--> 267     res = parse_datetime_string_with_reso(arg, freq=freq,
    268                                           dayfirst=dayfirst,
    269                                           yearfirst=yearfirst)

~/pandas/pandas/_libs/tslibs/parsing.pyx in pandas._libs.tslibs.parsing.parse_datetime_string_with_reso()
    291 
    292     if not _does_string_look_like_datetime(date_string):
--> 293         raise ValueError('Given date string not likely a datetime.')
    294 
    295     parsed, reso = _parse_delimited_date(date_string, dayfirst)

ValueError: Given date string not likely a datetime.
```

but the output should be `Series([False, False])`.  The other comparison operators `<`, `<=`, `>`, and `>=` yield the same error, which also seems incorrect (the error message should probably be different, and the error should be a `TypeError` rather than a `ValueError`).  This is happening on 0.25.1 and master.

### PR DESCRIPTION
**Pull request by OlivierLuG**
I've added test to compare `pd.Series([...], dtype="period[A-DEC]")`  to dtype `str`, `bool`, `int`, `float` and `None`.

- [ x ] closes #28980
- [ 5 ] tests added / passed
- [ x ] passes `black pandas`
- [ x ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by dsaxton** [comment]
Looks pretty good, can you modify the description with "closes" in front of the issue number so that it'll be closed automatically when this is merged?

**Comment by dsaxton** [comment]
> can you modify the description with "closes" in front of the issue number

@OlivierLuG I meant the description of the PR (where you have the bulleted list) rather than the PR title

File: pandas/tests/series/test_period.py [hunk 1]
```diff
@@ -113,3 +113,13 @@ def test_to_period(self, input_vals):
         expected = Series([input_vals], dtype="Period[D]")
         result = Series([input_vals], dtype="datetime64[ns]").dt.to_period("D")
         tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize("comparison_data", ["a", False, 1, 1.0, None])
```
Comment (jbrockmendel) [inline pandas/tests/series/test_period.py:117]: can you call this "other" or "scalar"

test belongs in tests.arithmetic.test_period

File: pandas/tests/series/test_period.py [hunk 2]
```diff
@@ -113,3 +113,13 @@ def test_to_period(self, input_vals):
...
+    def test_comparison_operations(self, comparison_data):
+        # GH 28980
+        expected = Series([False, False])
+        result = (
+            Series([pd.Period("2019"), pd.Period("2020")], dtype="period[A-DEC]")
```
Comment (jbrockmendel) [inline pandas/tests/series/test_period.py:122]: can you separate this into two lines like

```
ser = pd.Series(...)
result = ser == comparison_data
```

Comment (OlivierLuG) [reply pandas/tests/series/test_period.py]: Sure, it is done. I've push an update.

**Comment by OlivierLuG** [comment]
Hello, If you think this is correct, I can pull a new request to close #28981, which test about the same tests ?

**Comment by jreback** [comment]
thanks

File: pandas/tests/arithmetic/test_period.py [hunk 3]
```diff
@@ -1522,3 +1522,11 @@ def test_pi_sub_period_nat(self):
         exp = pd.TimedeltaIndex([np.nan, np.nan, np.nan, np.nan], name="idx")
         tm.assert_index_equal(idx - pd.Period("NaT", freq="M"), exp)
         tm.assert_index_equal(pd.Period("NaT", freq="M") - idx, exp)
+
+    @pytest.mark.parametrize("scalars", ["a", False, 1, 1.0, None])
+    def test_comparison_operations(self, scalars):
```
Comment (jbrockmendel) [inline pandas/tests/arithmetic/test_period.py:1527]: why not use the `box_with_array` fixture?

**Pull request merged**